identity_agent/src/didcomm/agent.rs

```diff
@@ -0,0 +1,294 @@
+    self.agent.start_listening(address).await
+  }
+
+  /// See [`Agent::agent_id`].
+  pub fn agent_id(&self) -> AgentId {
+    self.agent.agent_id()
+  }
+
+  /// See [`Agent::addresses`].
+  pub async fn addresses(&mut self) -> AgentResult<Vec<Multiaddr>> {
+    self.agent.addresses().await
+  }
+
+  /// See [`Agent::add_agent_address`].
+  pub async fn add_agent_address(&mut self, agent_id: AgentId, address: Multiaddr) -> AgentResult<()> {
+    self.agent.add_agent_address(agent_id, address).await
+  }
+
+  /// See [`Agent::add_agent_addresses`].
+  pub async fn add_agent_addresses(&mut self, agent_id: AgentId, addresses: Vec<Multiaddr>) -> AgentResult<()> {
+    self.agent.add_agent_addresses(agent_id, addresses).await
+  }
+
+  /// See [`Agent::shutdown`].
+  pub async fn shutdown(self) -> AgentResult<()> {
+    self.agent.shutdown().await
+  }
+
+  /// See [`Agent::send_request`].
+  pub async fn send_request<REQ: HandlerRequest>(
+    &mut self,
+    agent_id: AgentId,
+    request: REQ,
+  ) -> AgentResult<REQ::Response> {
+    self.agent.send_request(agent_id, request).await
+  }
+
+  /// Sends an asynchronous DIDComm request to an agent.
+  ///
+  /// To receive a possible response, call [`DidCommAgent::await_didcomm_request`] with the same `thread_id`.
+  pub async fn send_didcomm_request<REQ: DidCommRequest>(
+    &mut self,
+    agent_id: AgentId,
+    thread_id: &ThreadId,
+    message: REQ,
+  ) -> AgentResult<()> {
+    let endpoint: Endpoint = REQ::endpoint();
+    let request_mode: RequestMode = REQ::request_mode();
+
+    let dcpm = DidCommPlaintextMessage::new(thread_id.to_owned(), endpoint.to_string(), message);
+
+    self.create_thread_channels(thread_id);
+
+    let dcpm_vec = serde_json::to_vec(&dcpm).map_err(|err| Error::SerializationFailure {
+      location: ErrorLocation::Local,
+      context: "send message".to_owned(),
+      error_message: err.to_string(),
+    })?;
+
+    log::debug!("sending DIDComm request on endpoint `{endpoint}`");
+
+    let message: RequestMessage = RequestMessage::new(endpoint, request_mode, dcpm_vec);
+
+    let response = self.commander_mut().send_request(agent_id, message).await?;
+
+    serde_json::from_slice::<Result<(), RemoteSendError>>(&response.0).map_err(|err| {
+      Error::DeserializationFailure {
+        location: ErrorLocation::Local,
+        context: "send message".to_owned(),
+        error_message: err.to_string(),
+      }
+    })??;
+
+    Ok(())
+  }
+
+  /// Wait for a message on a given `thread_id`. This can only be called successfully if
+  /// [`DidCommAgent::send_didcomm_request`] was called on the same `thread_id` previously.
+  ///
+  /// This will return a timeout error if no message is received within the duration passed
+  /// to [`DidCommAgentBuilder::timeout`](crate::didcomm::DidCommAgentBuilder::timeout).
+  pub async fn await_didcomm_request<T: DeserializeOwned + Send + 'static>(
```

Comment:
> This can only be called successfully if
  /// [`DidCommAgent::send_didcomm_request`] was called on the same `thread_id` previously. 

Should the docs also mention what to expect if `DidCommAgent::send_didcomm_request` was called multiple times on the same `ThreadId` previously?

Reply:
> Calling `send_didcomm_request` multiple times still only allows to await one message on the thread.

Added this. We'll see during DIDComm protocol implementations whether this is too restrictive and can change it then if we need it.